common/frame.go

```diff
@@ -592,23 +591,53 @@ func (f *Frame) waitForFunction(
 
 	pageFn := `
 		(injected, predicate, polling, timeout, ...args) => {
-			return injected.waitForPredicateFunction(predicate, polling, timeout, ...args);
+			const fn = (...args) => {
+				return predicate(...args) || injected.continuePolling;
+			}
```

Comment:
I started adding a comment, but explaining this would require explaining internal details of `injected_script.js`.

So instead I just moved this to `injected_script.js` in #303, which hopefully shouldn't require any comments. Let me know.